pkg/block/block.go

```diff
@@ -126,6 +129,35 @@ func cleanUp(logger log.Logger, bkt objstore.Bucket, id ulid.ULID, err error) er
 	return err
 }
 
+// MarkForDeletion creates a file which stores information about when the block was marked for deletion.
+func MarkForDeletion(ctx context.Context, logger log.Logger, bkt objstore.Bucket, id ulid.ULID, blocksMarkedForDeletion prometheus.Counter) error {
+	deletionMarkFile := path.Join(id.String(), metadata.DeletionMarkFilename)
+	deletionMarkExists, err := objstore.Exists(ctx, bkt, deletionMarkFile)
+	if err != nil {
+		return errors.Wrapf(err, "check exists: %s in bucket", deletionMarkFile)
+	}
+	if deletionMarkExists {
+		return errors.Wrapf(err, "file %s already exists in bucket", deletionMarkFile)
+	}
+
+	deletionMark, err := json.Marshal(metadata.DeletionMark{
+		ID:           id,
+		DeletionTime: time.Now().Unix(),
+		Version:      metadata.DeletionMarkVersion1,
+	})
+	if err != nil {
+		return errors.Wrap(err, "json encode deletion mark")
+	}
+
+	if err := bkt.Upload(ctx, deletionMarkFile, bytes.NewReader(deletionMark)); err != nil {
+		return errors.Wrapf(err, "upload file: %s to bucket", deletionMarkFile)
+	}
+
+	blocksMarkedForDeletion.Inc()
```

Comment:
Perhaps the caller, instead of passing this counter, could increase its own counter if `MarkForDeletion` returns nil? It feels little weird to have extra counter argument to this function.

Reply:
I agree with this. Thank you for the feedback. I have made the required changes.